xbmc/interfaces/Builtins.cpp

```diff
@@ -1593,7 +1592,17 @@ int CBuiltins::Execute(const std::string& execString)
     if (params.size() > 1)
       singleFile = StringUtils::EqualsNoCase(params[1], "true");
     else
-      singleFile = !CGUIDialogYesNo::ShowAndGetInput(CVariant{iHeading}, CVariant{20426}, cancelled, CVariant{20428}, CVariant{20429});
+    {
+      CVariant options{CVariant::VariantTypeObject};
+      options["heading"] = iHeading;
+      options["text"] = 20426;
+      options["yeslabel"] = 20429;
+      options["nolabel"] = 20428;
+      
+      int result = CApplicationMessenger::Get().SendMsg(TMSG_GUI_DIALOG_YESNO, -1, -1, static_cast<void*>(&options));
```

Comment:
does it make sense to introduce another overload of that method to get rid of the `-1, -1` parameters in the case we only need the payload parameter.

```
void SendMsg(uint32_t messageId, void* payload = nullptr);
```